src/components/form/PillInput.res

```diff
@@ -0,0 +1,179 @@
+    }
+
+    if duplicateCheck && Array.some(items, existingItem => existingItem == trimmedItem) {
+      setError(_ => "Email already exists")
+      setSuggestion(_ => None)
+    } else if !CommonAuthUtils.isValidEmail(trimmedItem) {
+      setItems(prev => Array.concat(prev, [trimmedItem]))
+      form.change(name, [...items, elem]->Identity.genericTypeToJson)
+      setInputValue(_ => "")
+      setError(_ => "")
+      setSuggestion(_ => None)
+    } else {
+      setError(_ => "Invalid Email")
+      setSuggestion(_ => None)
+    }
+  }
+
+  let handleSuggestionClick = () => {
+    switch suggestion {
+    | Some(suggestedEmail) => addItem(suggestedEmail)
+    | None => ()
+    }
+  }
+
+  let removeItem = itemToRemove => {
+    form.change(name, items->Array.filter(ele => ele !== itemToRemove)->Identity.genericTypeToJson)
+    setItems(prev => prev->Array.filter(item => item != itemToRemove))
+  }
+
+  let saveItem = itemToSave => {
+    let trimmedEditInput = String.trim(editInput)
+    if (
+      duplicateCheck &&
+      trimmedEditInput != itemToSave &&
+      Array.some(items, existingItem => existingItem == trimmedEditInput)
+    ) {
+      setError(_ => "Email already exists")
+      setEditInput(_ => itemToSave)
+    } else if !CommonAuthUtils.isValidEmail(trimmedEditInput) {
+      setItems(prev => prev->Array.map(item => item == itemToSave ? trimmedEditInput : item))
+      let updatedArray =
+        items
+        ->Array.filter(ele => ele !== itemToSave)
+        ->Array.concat([trimmedEditInput])
+      form.change(name, updatedArray->Identity.genericTypeToJson)
+      setEditingItem(_ => None)
+      setEditInput(_ => "")
+      setError(_ => "")
+    } else {
+      setEditInput(_ => itemToSave)
+      setError(_ => "Invalid Email")
+    }
+  }
+
+  let handleKeyDown = e => {
+    let key = e->ReactEvent.Keyboard.key
+    let keyCode = e->ReactEvent.Keyboard.keyCode
+    if key === "Enter" || keyCode === 13 {
+      ReactEvent.Keyboard.preventDefault(e)
+      switch suggestion {
+      | Some(suggestedEmail) => addItem(suggestedEmail)
+      | None => addItem(inputValue)
+      }
+    }
+  }
+
+  let toggleEditingItem = item => {
+    setEditingItem(_ => Some(item))
+    setEditInput(_ => item)
+  }
+
+  <div className="w-full ">
+    <div className="w-full flex flex-wrap gap-2 border p-2 text-sm rounded-md">
+      {items
+      ->Array.mapWithIndex((item, i) =>
+        switch editingItem {
+        | Some(selected) if selected == item =>
+          <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border text-sm rounded-md">
+            <input
+              type_="text"
+              value={editInput}
+              onBlur={_ => saveItem(item)}
+              onInput=handleEditChange
+              onKeyDown={ev => {
+                let key = ev->ReactEvent.Keyboard.key
+                let keyCode = ev->ReactEvent.Keyboard.keyCode
+                if key === "Enter" || keyCode === 13 {
```

Comment:
and "Enter" and 13 these constants can be stored in the variable on used , its used another key handling function as well